lib/nerves_hub_link/socket.ex

```diff
@@ -388,15 +384,25 @@ defmodule NervesHubLink.Socket do
   end
 
   @impl Slipstream
-  def handle_info(:connect_check_network_availability, socket) do
-    case :inet.gethostbyname(to_charlist(socket.assigns.config.device_api_host)) do
-      {:ok, _} ->
+  def handle_info(:check_network_and_time, socket) do
+    cond do
+      network_connected?(socket) && time_synchronized?(socket) ->
         {:noreply, socket, {:continue, :connect}}
 
-      _ ->
-        Logger.info("[NervesHubLink] waiting for network to become available")
-        schedule_network_availability_check(2_000)
-        {:noreply, socket}
+      !network_connected?(socket) && !time_synchronized?(socket) ->
+        network_check_with_message(
+          "[NervesHubLink] waiting for network to become available and time to sync",
+          socket
+        )
```

Comment:
Do we need to be this verbose? As in reporting every possible condition vs just the first of time or connect problem we detect?

Reply:
Thats a good point, and I agree, and yet I flip flop.

I think you are right though. I'll change this around.